alcs-api/src/application/application.service.ts

```diff
@@ -57,26 +66,82 @@ export class ApplicationService {
     @InjectRepository(Application)
     private applicationRepository: Repository<Application>,
     private applicationTimeTrackingService: ApplicationTimeTrackingService,
+    private codeService: CodeService,
     @InjectMapper() private applicationMapper: Mapper,
   ) {}
 
-  async createOrUpdate(
-    application: Partial<ApplicationUpdateServiceDto>,
-  ): Promise<Application> {
-    let existingApplication = await this.applicationRepository.findOne({
+  async create(application: CreateApplicationDto): Promise<Application> {
+    const existingApplication = await this.applicationRepository.findOne({
       where: { fileNumber: application.fileNumber },
     });
 
+    if (existingApplication) {
+      throw new ServiceValidationException(
+        'Application with file number already exists',
+      );
+    }
+
+    const type = await this.codeService.fetchApplicationType(application.type);
+
+    const region = application.region
+      ? await this.codeService.fetchRegion(application.region)
+      : undefined;
+
+    const newApplication = new Application({
+      fileNumber: application.fileNumber,
+      applicant: application.applicant,
+      dateReceived: new Date(application.dateReceived),
+      localGovernmentUuid: application.localGovernmentUuid,
+      type,
+      region,
+    });
+
+    newApplication.card = new Card();
+    await this.applicationRepository.save(newApplication);
+
+    return this.get(application.fileNumber);
+  }
+
+  async updateByFileNumber(
+    fileNumber: string,
+    updates: ApplicationUpdateServiceDto,
+  ) {
+    const existingApplication = await this.applicationRepository.findOne({
+      where: { fileNumber },
+    });
+
+    if (!existingApplication) {
+      throw new ServiceNotFoundException(
+        `Application not found with file number ${fileNumber}`,
+      );
+    }
+
+    return this.update(existingApplication, updates);
+  }
+
+  async updateByUuid(uuid: string, updates: ApplicationUpdateServiceDto) {
+    const existingApplication = await this.applicationRepository.findOne({
+      where: { uuid },
+    });
+
     if (!existingApplication) {
-      existingApplication = new Application();
-      existingApplication.card = new Card();
+      throw new ServiceNotFoundException(
+        `Application not found with file number ${uuid}`,
+      );
     }
 
-    const updatedApp = Object.assign(existingApplication, application);
+    return this.update(existingApplication, updates);
+  }
+
+  async update(
+    existingApplication: Application,
+    updates: ApplicationUpdateServiceDto,
+  ): Promise<Application> {
+    //TODO: Actual Update Logic
```

Comment:
is this still relevant

Reply:
oh snap, maybe! I was thinking of moving away from object.assign, Ill go back to it.

Reply:
Reviewed how we are using this, since its using its own specific type that is'nt the same as the controller, I think using Object.assign is okay.